### ISSUE TITLE
Feedback from a first time user of this tool (sen. dev)

### ISSUE THREAD

**Issue by WernerMairl** [natemcmaster/CommandLineUtils#194]
Hi
i'm using your tool (and i like it!) the first time, so i have some suggestions about usability.....

## Finding 1
```
CommandLineApplication.Execute<XXX>()
```

seems to execute a Method "OnExecute" on class XXX by Name (reflection !?)
The only hint that i found about this was inside the intellisense tooltip (=> increase visibility of how the feature works)
for me this is a core concept, that can be better presented in the documentation (method must be public or private ? Naming Convention etc...)

using the following rows 
```
app = new CommandLineApplication<XXX>()
// app.OnExecute ( () => ) explicitely NOT called/implemented
app.execute(args);
```

the "OnExecute" Method on class XXX seems to not be called.
Bug or Intended ?
in every case... improvement of documentation/visibility suggested

## Finding 2:

using the following way, it was "hard" to find the Instance of XXX

```
app = new CommandLineApplication<XXX>()
```

yes, i have found it as

```
 XXX x = app.Model;
```
using "Ctrl+J" in VS you must scroll down a lot and "Model" is not the first wording that me comes in mind ... my first try was something that starts with "A" like Application... but there is nothing

suggestions: think about rename Property "Model" into "Application" or something else that helps...

## Bottom Line:
only a few suggestions about usabilty from someone new on this API.
Feel free to close this issue/create single one for my suggestions etc....

regards
Werner
